### PR TITLE
Fix for NumAdaptivityCycles postprocessor when manually turning off mesh adaptivity

### LINKED ISSUE #9746: NumAdaptivityCycles Postprocessor reports wrong value when mesh adaptivity is manually disabled
### Description of the enhancement or error report
When `Adaptivity::setAdaptivityOn(false)` is called, the adaptivity cycle counter is never reset ([here](https://github.com/idaholab/moose/blob/devel/framework/src/base/FEProblemBase.C#L4451)) because the whole `FEProblemBase::adaptMesh()` method gets skipped completely on subsequent time steps.

### Rationale for the enhancement or information for reproducing the error
Use the `NumAdaptivityCycles` postprocessor in a problem where `Adaptivity::setAdaptivityOn(false) `is called any time after some mesh refinement has already occurred.

### Identified impact
(i.e. Internal object changes, limited interface changes, public API change, or a list of specific applications impacted)
Slight improvement to test objects.

I will submit a PR for this. It will involve a tiny refactor to the adaptivity logic. I think I have a good test for this too.

### PR DESCRIPTION
**Pull request by jessecarterMOOSE**
Slightly modifies the mesh adaptivity logic so that `FEProblemBase::adaptMesh()` always gets run, which allows the adaptivity cycle counter to be reset, allowing `FEProblemBase::getNumCyclesCompleted()` to return to correct value if mesh adaptivity gets shut off manually (ie, calling `Adaptivity::setAdaptivityOn(false)`).

I removed what seemed like a redundant mesh adaptivity check, which was occurring both before `FEProblemBase::adaptMesh()` was called and in the first line of it (different methods were being called but they both require `mesh_refinement_on` to be true). I just removed the first check so the cycle counter would always get reset, and then the `adaptMesh()` function could be skipped from within if necessary.

I also added a couple tests to demonstrate the functionality.

Closes #9746

### CODE REVIEW

**Comment by jessecarterMOOSE** [comment]
Hmm, I fixed one of my commits, squashed it in with the formatting fix, and did a force push. Now some of CIVET tests aren't running. What should I do?

File: test/src/userobjects/ToggleMeshAdaptivity.C [hunk 1]
```diff
@@ -52,5 +58,6 @@ ToggleMeshAdaptivity::initialSetup()
 void
 ToggleMeshAdaptivity::execute()
 {
-  checkState();
+  if (_t_step > _steps_to_wait)
```
Comment (rwcarlsen) [inline test/src/userobjects/ToggleMeshAdaptivity.C]: Should be ``>=`` to allow waiting for zero steps.

**Review by rwcarlsen — COMMENTED** [review]
Looks good. @permcody or @jwpeterson probably should give the final okay on the adaptivity logic mod.

File: test/src/userobjects/ToggleMeshAdaptivity.C (hunk 1 above)
Comment (permcody) [reply test/src/userobjects/ToggleMeshAdaptivity.C]: Yeah probably, It's just a test object though so no real impact of choosing one method over another.

**Review by permcody — APPROVED** [review]

**Pull request merged**